packages/@aws-cdk/aws-rds/lib/cluster.ts

```diff
@@ -606,13 +606,20 @@ export class DatabaseCluster extends DatabaseClusterBase {
   }
 
   private setLogRetention(props: DatabaseClusterProps) {
-    if (props.cloudwatchLogsExports && props.cloudwatchLogsRetention) {
-      for (const log of props.cloudwatchLogsExports) {
-        new lambda.LogRetention(this, `LogRetention${log}`, {
-          logGroupName: `/aws/rds/cluster/${this.clusterIdentifier}/${log}`,
-          retention: props.cloudwatchLogsRetention,
-          role: props.cloudwatchLogsRetentionRole,
-        });
+    if (props.cloudwatchLogsExports) {
+      const unsupportedLogNames = props.cloudwatchLogsExports.filter(logName => !props.engine.logTypes.includes(logName));
```

Comment:
I feel like this code just begs for a utility method:

```ts
      const unsupportedLogNames = props.cloudwatchLogsExports.filter(logName => !engineSupportsLogType(props.engine, logName));

  function engineSupportsLogType(engine: IClusterEngine, logType: string): boolean {
     // see my comment above about this being optional
     if (!engine.supportedLogTypes) { return true; }
     return engine.supportedLogTypes.includes(logType);
  }
```

Reply:
I can agree if we decide to make the `supportedLogTypes` optional; otherwise, it's not (in my mind) noticeably improving the readability of the code.